src/machine.rs

```diff
@@ -422,6 +427,34 @@ impl<'m> Machine<'m> {
 
         Ok(())
     }
+
+    #[instrument(skip_all)]
+    async fn pre_start_cleanup(&self) -> Result<(), Error> {
```

Comment:
For some reason I do not like the name `pre_start_cleanup`.. idk maybe `cleanup_before_start` is a bit easier to read?